spec/data/sig_handler.rb.txt

```diff
@@ -165,7 +165,7 @@ class CollectionSigs
   sig{params(arr: Custom[String]).void}
   def custom_collection(arr); end
 
-  sig{params(arr: T::Array[Custom[String]]).void}
+  sig{params(arr: T::Array[Custom1[Custom2[String]]]).void}
```

Comment:
Can we stress test this a little more?
```suggestion
  sig{params(arr: T::Array[Custom1[Custom2[String, Integer, T::Boolean], T.any(Custom3[String], Custom4[Integer])]]).void}
```

Basically testing this:
```ruby
T::Array[
  Custom1[
    Custom2[
      String,
      Integer, 
      T::Boolean
    ], 
    T.any(
      Custom3[String],
      Custom4[Integer]
    )
  ]
]
```
Not that we expect anyone to write anything like this, but would be nice to have a complex test case for this.